### DISCUSSION TITLE
Something wrong about using JUDI

### DISCUSSION THREAD

**Discussion by venividivici666** [slimgroup/JUDI.jl#234] [General]
When I used JUDI, there is a problem:
julia> using JUDI
ERROR: InitError: PyError (PyImport_ImportModule

The Python package interface could not be imported by pyimport. Usually this means
that you did not install interface in the Python version being used by PyCall.

PyCall is currently configured to use the Python version at:

/home/brcgpt/Anaconda/anaconda/bin/python

and you should use whatever mechanism you usually use (apt-get, pip, conda,
etcetera) to install the Python package containing the interface module.

One alternative is to re-configure PyCall to use a different Python
version on your system: set ENV["PYTHON"] to the path/name of the python
executable you want to use, run Pkg.build("PyCall"), and re-launch Julia.

Another alternative is to configure PyCall to use a Julia-specific Python
distribution via the Conda.jl package (which installs a private Anaconda
Python distribution), which has the advantage that packages can be installed
and kept up-to-date via Julia.  As explained in the PyCall documentation,
set ENV["PYTHON"]="", run Pkg.build("PyCall"), and re-launch Julia. Then,
To install the interface module, you can use `pyimport_conda("interface", PKG)`,
where PKG is the Anaconda package that contains the module interface,
or alternatively you can use the Conda package directly (via
`using Conda` followed by `Conda.add` etcetera).

) <class 'ModuleNotFoundError'>
ModuleNotFoundError("No module named 'pyrevolve'")
  File "/home/brcgpt/.julia/packages/JUDI/9qfzJ/src/pysource/interface.py", line 5, in <module>
    from pyrevolve import Revolver

And I can't fix it. Would you help me? Thanks a lot!

**Comment by mloubout** [comment]
Looks like `pyrevolve` isn't installed, how did you install JUDI and devito?

**Reply by venividivici666** [reply]
Install JUDI:
] add JUDI

Install Devito:
git clone https://github.com/devitocodes/devito.git
cd devito
pip install -e .

**Reply by mloubout** [reply]
You need `pip install -e .[extras,tests]` to get all needed dependencies.

You will also need to make sure julia uses your system python if you install devito yourself i.e

```bash
PYTHON=$(which python3) julia -e 'using Pkg;Pkg.build(["PyCall", "JUDI"])'
```
